scandium-core/src/main/java/org/eclipse/californium/scandium/DTLSConnector.java

```diff
@@ -1511,12 +1527,26 @@ private void startNewHandshake(final ClientHello clientHello, final Record recor
 	 * @throws HandshakeException if the session cannot be resumed based on the parameters
 	 *             provided in the client hello message
 	 */
-	private void resumeExistingSession(final ClientHello clientHello, final Record record, final Connection connection) throws HandshakeException {
-		InetSocketAddress peerAddress = connection.getPeerAddress();
+	private void resumeExistingSession(ClientHello clientHello, Record record, final AvailableConnections connections)
+			throws HandshakeException {
+		InetSocketAddress peerAddress = record.getPeerAddress();
 		LOGGER.debug("Client [{}] wants to resume session with ID [{}]", peerAddress, clientHello.getSessionId());
 
+		if (connections == null) {
+			throw new NullPointerException("available connections must not be null!");
+		}
+		Connection connection = connections.getConnectionByAddress();
+		if (connection == null) {
+			throw new NullPointerException("connection by address must not be null!");
+		} else if (!connection.equalsPeerAddress(peerAddress)) {
+			throw new IllegalArgumentException("connection must have records address!");
+		}
+
 		SessionTicket ticket = null;
-		final Connection previousConnection = connectionStore.find(clientHello.getSessionId());
+		if (!connections.isSetConnectionBySessionId()) {
+			connections.setConnectionBySessionId(connectionStore.find(clientHello.getSessionId()));
+		}
+		Connection previousConnection = connections.getConnectionBySessionId();
```

Comment:
I'm not sure to understand why we set `connections` as we never use it after ?
It could be :
``` java
Connection previousConnection;
if (connections.isSetConnectionBySessionId()) {
    previousConnection = connections.getConnectionBySessionId(); 
} else {
    previousConnection = connectionStore.find(clientHello.getSessionId());
}
```

Reply:
Is the question still pending?
The intention is to limit the calls to `find`, even if `find` has returned `null`.

Reply:
> I guess, it's more or less a different opinion on simpleness :-)

Yes :)